src/Microsoft.DotNet.Interactive/Parsing/SubmissionParser.cs

```diff
@@ -124,7 +124,7 @@ public IReadOnlyList<KernelCommand> SplitSubmission(RequestDiagnostics requestDi
                                         CodeAnalysis.DiagnosticSeverity.Error,
                                         "NI0001", // QUESTION: (SplitSubmission) what code should this be?
                                         "Unrecognized magic command");
-                                    var diagnosticsProduced = new DiagnosticsProduced(new[] { diagnostic }, c);
+                                    var diagnosticsProduced = new DiagnosticsProduced(new[] { diagnostic }, originalCommand);
```

Comment:
There were some inadvertent whitespace fix ups - but this is the main part of the fix.